mesonbuild/modules/python.py

```diff
@@ -727,10 +727,23 @@ def find_installation(self, state: 'ModuleState', args: T.Tuple[T.Optional[str]]
         found_modules: T.List[str] = []
         missing_modules: T.List[str] = []
         if python.found() and want_modules:
+            # Python 3.8.x or later require add_dll_directory() to be called on Windows if
+            # the needed modules require external DLLs that are not bundled with the modules.
+            # Simplify things by calling add_dll_directory() on the paths in %PATH%
+            add_paths_cmd = ''
+            if hasattr(os, 'add_dll_directory'):
+                add_paths_cmds = []
+                paths = os.environ['PATH'].split(os.pathsep)
+                for path in paths:
+                    if path != '' and os.path.isdir(path):
+                        add_paths_cmds.append('os.add_dll_directory({!r})'.format(path))
+                add_paths_cmds.reverse()
+                add_paths_cmd = 'import os;' + ';'.join(add_paths_cmds) + ';'
```

Comment:
```suggestion
                add_paths_cmd = 'import os;' + ';'.join(reversed(add_paths_cmds)) + ';'
```

Or alternatively you might wish to do `for path in reversed(paths):`  up above.